### PR TITLE
changes for static compilation

### PR DESCRIPTION
**Pull request by SimonDanisch**
mostly turned globals into constants with a concrete type.

### CODE REVIEW

File: src/core.jl [hunk 1]
```diff
@@ -52,32 +71,31 @@ else
...
 
-const edges = Vector{Int}[] #parents to children, useful for plotting graphs
 
-const node_count = DefaultDict{String,Int}(0) #counts of different signals for naming
 function auto_name!(name, parents...)
     parents_str = join(map(s->s.name, parents), ", ")
     isempty(parents_str) || (name *= "($parents_str)")
-    node_count[name] += 1
-    countstr = node_count[name] > 1 ? "-$(node_count[name])" : ""
+    count = get!(node_count, name, 0) + 1
```
Comment (JobJob) [inline src/core.jl]: `node_count` is a DefaultDict so don't need the `get!` But you could just change it to a regular Dict, and leave the `get!`, if you prefer that style.

Comment (SimonDanisch) [reply src/core.jl]: That's what I did ;)

Comment (JobJob) [reply src/core.jl]: Ha lol, ok

Comment (SimonDanisch) [reply src/core.jl]: To be honest, I'm not a hundred percent sure if that was a problem. But globals turned out pretty fragile, so I tried to use only Base types for globals. And the stack traces pointed to auto_name! and this was, with other changes, the first one working.

Comment (JobJob) [reply src/core.jl]: Sweet

**Comment by JobJob** [comment]
So there's nothing here besides formatting, that node_count change, and runner_task changed to a ref? If so, looks good to me.

**Pull request merged**